llvm/lib/Target/X86/GISel/X86InstructionSelector.cpp

```diff
@@ -1789,6 +1793,47 @@ bool X86InstructionSelector::selectMulDivRem(MachineInstr &I,
   return true;
 }
 
+bool X86InstructionSelector::selectSelect(MachineInstr &I,
+                                          MachineRegisterInfo &MRI,
+                                          MachineFunction &MF) const {
+  unsigned DstReg = I.getOperand(0).getReg();
+  BuildMI(*I.getParent(), I, I.getDebugLoc(), TII.get(X86::TEST32rr))
+      .addReg(I.getOperand(1).getReg())
+      .addReg(I.getOperand(1).getReg());
+
```

Comment:
Amara mentioned it in the FCMP PR:
```
auto &Select = cast<GSelect>(I);
```
It will give you access to all the Registers.